test/OrchardCore.Tests/Modules/OrchardCore.Tenants/Services/TenantValidatorTests.cs

```diff
@@ -0,0 +1,124 @@
+                RequestUrlPrefix = urlPrefix,
+                RequestUrlHost = hostName,
+                FeatureProfile = featureProfile
+            };
+
+            var errors = await tenantValidator.ValidateAsync(viewModel);
+
+            // Assert
+            Assert.Equal(errorMessages.Length, errors.Count());
+
+            for (var i = 0; i < errors.Count(); i++)
+            {
+                Assert.Equal(errorMessages[i], errors.ElementAt(i).Message);
+            }
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ShouldNotCreateTenantAlreadyUsedInMultipleHostnames(bool isNewTenant)
+        {
+            // Arrange
+            var tenantValidator = CreateTenantValidator();
+
+            var viewModel = new EditTenantViewModel
+            {
+                Name = "Tenant2",
+                RequestUrlHost = "example2.com",
+                FeatureProfile = "Feature Profile",
+                IsNewTenant = isNewTenant
+            };
+
+            _shellSettings.Add(new ShellSettings
+            {
+                Name = "Tenant1",
+                RequestUrlHost = "exmple1.com, example2.com"
+            });
+
+            // Act
+            var errors = await tenantValidator.ValidateAsync(viewModel);
+
+            // Asserts
+            Assert.Single(errors);
+            Assert.Equal("A tenant with the same host and prefix already exists.", errors.Single().Message);
+        }
+
+        private TenantValidator CreateTenantValidator(bool defaultTenant = true)
+        {
+            var shellHostMock = new Mock<IShellHost>();
+            shellHostMock.Setup(h => h.GetAllSettings()).Returns(_shellSettings);
+
+            var featureProfilesServiceMock = new Mock<IFeatureProfilesService>();
+            featureProfilesServiceMock.Setup(fp => fp.GetFeatureProfilesAsync())
+                .Returns(Task.FromResult((IDictionary<string, FeatureProfile>)new Dictionary<string, FeatureProfile>
+                {
+                    { "Feature Profile", new FeatureProfile() }
+                }));
+
+            var stringLocalizerMock = new Mock<IStringLocalizer<TenantValidator>>();
+            stringLocalizerMock
+                .Setup(l => l[It.IsAny<string>()])
+                .Returns<string>(n => new LocalizedString(n, n));
+            stringLocalizerMock
+                .Setup(l => l[It.IsAny<string>(), It.IsAny<object[]>()])
+                .Returns<string, object[]>((n, a) => new LocalizedString(n, n));
+
+            var shellSettings = defaultTenant
+                ? _shellSettings.First()
+                : new ShellSettings();
+
+            return new TenantValidator(
+                shellHostMock.Object,
+                featureProfilesServiceMock.Object,
+                Enumerable.Empty<DatabaseProvider>(),
+                shellSettings,
+                stringLocalizerMock.Object);
+        }
+
+        private void SeedTenants()
```

Comment:
Add some valid host-prefix combinations here, and test against those in `ShouldNotCreateTenantAlreadyUsedInMultipleHostnames`. That way, you can test more variations there, not just a single one (also data-driven).